### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by Nahid1911**
### Project requirements
- Install CanCanCan in your project.
- Add a `role` column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

For more details about requiements please click [HERE](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_add_authorization.md)

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the user here. For example:
+    #
+    return unless user.present?
+
+    can :read, :all
```
Comment (mikethreels) [inline app/models/ability.rb]: - [ ] setting your app up like this the user that created the post or comment isn't able to delete it.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -10,26 +10,35 @@ def show
     @posts = Post.all
     @post = Post.find params[:id]
     @post_id = @posts.index(@post) + 1
+    @user = @post.author
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb]: - [ ] according to the requirements your application should use CanCanCan for Post authorization in Post Controller

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job setting up the authorization

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/ability.rb (hunk 1 above)
Comment (Nahid1911) [reply app/models/ability.rb]: Resolved

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (Nahid1911) [reply app/controllers/posts_controller.rb]: Resolved

Comment (rloterh) [reply app/controllers/posts_controller.rb]: well resolved! ✔️

File: app/models/ability.rb (hunk 1 above)
Comment (rloterh) [reply app/models/ability.rb]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Muh & Nahid,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on introducing improvements to the authorization parameters!
🎲 Good ReadMe!

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**